src/merkle/tiered_smt/mod.rs

```diff
@@ -406,4 +461,22 @@ impl BottomLeaf {
         // TODO: hash in domain
         Rpo256::hash_elements(&elements)
     }
+
+    /// Returns contents of this leaf as a vector of (key, value) pairs.
+    ///
+    /// The keys are returned in their un-truncated form.
+    pub fn contents(&self, index: u64) -> Vec<(RpoDigest, Word)> {
```

Comment:
Could you provide some context around the purpose of `index` here please?  Why isn't it part of the `BottomLeaf` constructor?

Reply:
I would say that BottomLeaf should be self contained but I'm not strongly opinionated about this.